File: BinaryClock.java
import java.time.LocalTime;

public class BinaryClock {
	public static void main(String[] args) throws InterruptedException {
		while(true) {
			LocalTime time = LocalTime.now();
			int curSec = time.getSecond();
			int curMin = time.getMinute();
			int curHour = time.getMinute();
			
			String timeFormat = String.format("%s - %s - %s", Integer.toBinaryString(curHour), Integer.toBinaryString(curMin), Integer.toBinaryString(curSec));
			System.out.println(timeFormat);
			Thread.sleep(1000);
		}
	}
}

File: BinaryClock.kt
import java.time.LocalTime

fun main(args: Array<String>) {
    while (true) {
        val time: LocalTime = LocalTime.now()
        val curSec = time.second
        val curMin = time.minute
        val curHour = time.hour

        val timeFormat: String? = "%s - %s - %s".format(Integer.toBinaryString(curHour), Integer.toBinaryString(curMin), Integer.toBinaryString(curSec))
        println(timeFormat)
        Thread.sleep(1000)
    }
}
